tlair/src/main/scala/at/forsyte/apalache/tla/lir/Builder.scala

```diff
@@ -469,7 +469,9 @@ class Builder {
 
   /** TlaSetOper */
   def enumSet(args: BuilderEx*): BuilderEx = {
-    BuilderOper(TlaSetOper.enumSet, args: _*)
+    // Ensure we don't have duplicate elements in the set
+    val elems = args.toSet.toSeq
+    BuilderOper(TlaSetOper.enumSet, elems: _*)
```

Comment:
How do we check for semantic equality? I’m a bit confused why you approved this PR if it doesn’t fix the problem. :/

Reply:
Oh, never mind. It does solve the problem this issue is targeting. Maybe there are other forms of syntactically different values that could end up associated with duplicate indices, but That wasn’t the case in the bug reported.

also, if we can check semantic equality here, why can’t we also check for it when we are constructing the functions, to guard against duplicate indices in the first place?

Reply:
> 

We can only check semantic equality in SMT. Your change is the best effort to eliminate duplicates, when we can do that statically. For example, if you have a set `{ x, y }` of two state variables `x` and `y`, then you can neither eliminate `x`, nor `y`. However, it may happen that during execution `x` and `y` are equal. We can only see that by obtaining an SMT model.